src/utils.ts

```diff
@@ -26,3 +35,37 @@ const AsyncFunctionConstructor = (async () => {}).constructor;
  * an async function check method only consider runtime support async syntax
  */
 export const isAsyncFunction = (fn: Fn) => fn.constructor === AsyncFunctionConstructor;
+
+export const isAsyncTask = async (task: Task) => {
+  if (isAsyncFunction(task.fn)) {
+    return true;
+  }
+  try {
+    if (task.opts.beforeEach != null) {
+      try {
+        await task.opts.beforeEach.call(task);
+      } catch (e) {
```

Comment:
why do we ignore thrown errors here?

Reply:
Because this function is only responsible for determining whether this function is synchronous